In [1]:
#importing the libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.preprocessing import LabelEncoder
from sklearn.model_selection import train_test_split
from xgboost import XGBRegressor
from sklearn import metrics

In [2]:
#loading the dataset
big_mart_dt = pd.read_csv("C:/Users/MACHIRA/Projects_With_Siddardhan/data/big_mart_sale.csv")
big_mart_dt.head()

Item_Identifier  Item_Weight Item_Fat_Content  ...  Outlet_Location_Type        Outlet_Type  Item_Outlet_Sales
0           FDA15         9.30          Low Fat  ...                Tier 1  Supermarket Type1          3735.1380
1           DRC01         5.92          Regular  ...                Tier 3  Supermarket Type2           443.4228
2           FDN15        17.50          Low Fat  ...                Tier 1  Supermarket Type1          2097.2700
3           FDX07        19.20          Regular  ...                Tier 3      Grocery Store           732.3800
4           NCD19         8.93          Low Fat  ...                Tier 3  Supermarket Type1           994.7052

[5 rows x 12 columns]

In [3]:
#let's get the number of columns and rows
big_mart_dt.shape

(8523, 12)

In [4]:
#let's more info about the data set
big_mart_dt.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8523 entries, 0 to 8522
Data columns (total 12 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Item_Identifier            8523 non-null   object 
 1   Item_Weight                7060 non-null   float64
 2   Item_Fat_Content           8523 non-null   object 
 3   Item_Visibility            8523 non-null   float64
 4   Item_Type                  8523 non-null   object 
 5   Item_MRP                   8523 non-null   float64
 6   Outlet_Identifier          8523 non-null   object 
 7   Outlet_Establishment_Year  8523 non-null   int64  
 8   Outlet_Size                6113 non-null   object 
 9   Outlet_Location_Type       8523 non-null   object 
 10  Outlet_Type                8523 non-null   object 
 11  Item_Outlet_Sales          8523 non-null   float64
dtypes: float64(4), int64(1), object(7)
memory usage: 799.2+ KB


 #### Categorical Features:
    Item_Identifier            8523 non-null   object 
    Item_Fat_Content           8523 non-null   object 
    Item_Type                  8523 non-null   object 
    Outlet_Identifier          8523 non-null   object 
    Outlet_Size                6113 non-null   object 
    Outlet_Location_Type       8523 non-null   object 
    Outlet_Type                8523 non-null   object 


In [6]:
#let's check for missing values
big_mart_dt.isna().sum()

Item_Identifier                 0
Item_Weight                  1463
Item_Fat_Content                0
Item_Visibility                 0
Item_Type                       0
Item_MRP                        0
Outlet_Identifier               0
Outlet_Establishment_Year       0
Outlet_Size                  2410
Outlet_Location_Type            0
Outlet_Type                     0
Item_Outlet_Sales               0
dtype: int64